pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/PipInstallTask.groovy

```diff
@@ -134,81 +142,106 @@ class PipInstallTask extends DefaultTask implements FailureReasonProvider {
         def extension = ExtensionUtils.getPythonExtension(project)
         def sitePackages = findSitePackages()
 
-        for (File installable : getConfigurationFiles()) {
+        ProgressLoggerFactory progressLoggerFactory = getServices().get(ProgressLoggerFactory)
+        ProgressLogger progressLogger = progressLoggerFactory.newOperation(PipInstallTask)
+        progressLogger.setDescription("Installing Libraries")
+
+        PythonTag pythonTag = PythonTag.findTag(getProject(), getPythonDetails())
+        PlatformTag platformTag = PlatformTag.makePlatformTag(getProject(), getPythonDetails())
+        def cache = new WheelCache(wheelCache, pythonTag, platformTag)
+
+        progressLogger.started()
+        TaskTimer taskTimer = new TaskTimer()
+
+        int counter = 0
+        def installableFiles = getConfigurationFiles()
+        for (File installable : installableFiles) {
             if (isReadyForInstall(installable)) {
                 def packageInfo = PackageInfo.fromPath(installable.getAbsolutePath())
                 String shortHand = packageInfo.version ? "${packageInfo.name}-${packageInfo.version}" : packageInfo.name
 
-                if (packageExcludeFilter.isSatisfiedBy(packageInfo)) {
-                    logger.lifecycle(PythonHelpers.createPrettyLine("Install ${shortHand}", "[EXCLUDED]"))
-                    continue
-                }
-
-                String sanitizedName = packageInfo.name.replace('-', '_')
-
-                // See: https://www.python.org/dev/peps/pep-0376/
-                File egg = sitePackages.resolve("${sanitizedName}-${packageInfo.version}-py${pyVersion}.egg-info").toFile()
-                File dist = sitePackages.resolve("${sanitizedName}-${packageInfo.version}.dist-info").toFile()
-
-                def mergedEnv = new HashMap(extension.pythonEnvironment)
-                if (environment != null) {
-                    mergedEnv.putAll(environment)
-                }
-
-                if (project.file(egg).exists() || project.file(dist).exists()) {
-                    logger.lifecycle(PythonHelpers.createPrettyLine("Install ${shortHand}", "[SKIPPING]"))
```

Comment:
Do we still need this logging on some level to be able to debug/reason about it?

Reply:
Can you comment on this old question above?

Reply:
Yes. Otherwise in CI it' wont' show up, so we won't be able to debug what happened.